NoClassDefFoundError after

Hello,

I get this error after installing the `dmn-scala\zeebe-worker` :

`❯ java -jar .\target\dmn-engine-zeebe-worker-1.6.1-SNAPSHOT-full.jar

Exception in thread "main" java.lang.NoClassDefFoundError: org/apache/logging/log4j/util/StackLocator$FqcnCallerLocator
        at org.apache.logging.log4j.util.StackLocator.<clinit>(StackLocator.java:37)
        at org.apache.logging.log4j.util.StackLocatorUtil.<clinit>(StackLocatorUtil.java:33)
        at org.apache.logging.slf4j.Log4jLoggerFactory.getContext(Log4jLoggerFactory.java:44)
        at org.apache.logging.log4j.spi.AbstractLoggerAdapter.getLogger(AbstractLoggerAdapter.java:48)
        at org.apache.logging.slf4j.Log4jLoggerFactory.getLogger(Log4jLoggerFactory.java:30)
        at org.slf4j.LoggerFactory.getLogger(LoggerFactory.java:358)
        at org.camunda.dmn.package$.<clinit>(package.scala:7)
        at org.camunda.dmn.zeebe.ZeebeDmnWorkerApplication.start(ZeebeDmnWorkerApplication.scala:52)
        at org.camunda.dmn.zeebe.ZeebeDmnWorkerApplication$.main(ZeebeDmnWorkerApplication.scala:27)
        at org.camunda.dmn.zeebe.ZeebeDmnWorkerApplication.main(ZeebeDmnWorkerApplication.scala)
Caused by: java.lang.ClassNotFoundException: org.apache.logging.log4j.util.StackLocator$FqcnCallerLocator
        at java.base/jdk.internal.loader.BuiltinClassLoader.loadClass(BuiltinClassLoader.java:606)
        at java.base/jdk.internal.loader.ClassLoaders$AppClassLoader.loadClass(ClassLoaders.java:168)
        at java.base/java.lang.ClassLoader.loadClass(ClassLoader.java:522)
        ... 10 more
Exception in thread "shutdownHook1" java.lang.NoClassDefFoundError: Could not initialize class org.camunda.dmn.package$
        at org.camunda.dmn.zeebe.ZeebeDmnWorkerApplication.stop(ZeebeDmnWorkerApplication.scala:73)
        at org.camunda.dmn.zeebe.ZeebeDmnWorkerApplication$.$anonfun$main$3(ZeebeDmnWorkerApplication.scala:25)
        at scala.sys.ShutdownHookThread$.$anonfun$apply$1(ShutdownHookThread.scala:35)
        at java.base/java.lang.Thread.run(Thread.java:832)`


I also get the same error when I run `java -jar .\dmn-engine-zeebe-worker-1.6.0-full.jar`.

Is this repo being maintained or is it obsolete ?

Environment : 

* Windows 10
* Java version : 
`❯ java -version 
java version "15.0.2" 2021-01-19 
Java(TM) SE Runtime Environment (build 15.0.2+7-27) 
Java HotSpot(TM) 64-Bit Server VM (build 15.0.2+7-27, mixed mode, sharing) `

--- a/dmn-engine/src/main/scala/org/camunda/dmn/DmnEngine.scala
+++ b/dmn-engine/src/main/scala/org/camunda/dmn/DmnEngine.scala
@@ -2,14 +2,8 @@ package org.camunda.dmn
 
 import java.io.InputStream
 import java.util.ServiceLoader
-import org.camunda.dmn.Audit.{
-  AuditLog,
-  AuditLogEntry,
-  AuditLogListener,
-  ContextEvaluationResult,
-  EvaluationResult,
-  SingleEvaluationResult
-}
+
+import org.camunda.dmn.Audit._
 import org.camunda.dmn.evaluation._
 import org.camunda.dmn.parser._
 import org.camunda.feel.FeelEngine
@@ -117,11 +111,8 @@ class DmnEngine(configuration: DmnEngine.Configuration =
 
   val parser = new DmnParser(
     configuration = configuration,
-    parser = exp =>
-      feelEngine
-        .parseExpression(exp)
-        .left
-        .map(_.message)
+    parser = feelEngine.parseExpression(_).left.map(_.message),
+    unaryTestsParser = feelEngine.parseUnaryTests(_).left.map(_.message)
   )
 
   val decisionEval = new DecisionEvaluator(eval = this.evalExpression,

--- a/dmn-engine/src/main/scala/org/camunda/dmn/parser/DmnParser.scala
+++ b/dmn-engine/src/main/scala/org/camunda/dmn/parser/DmnParser.scala
@@ -21,7 +21,6 @@ import org.camunda.bpm.model.dmn.instance.{
 import org.camunda.dmn.DmnEngine.{Configuration, Failure}
 import org.camunda.feel.impl.parser.FeelParser
 import org.camunda.feel.syntaxtree._
-import org.camunda.feel.impl.parser.FeelParser.{NoSuccess, Success}
 
 import scala.collection.JavaConverters._
 import scala.collection.mutable
@@ -34,7 +33,8 @@ object DmnParser {
 }
 
 class DmnParser(configuration: Configuration,
-                parser: String => Either[String, ParsedExpression]) {
+                parser: String => Either[String, ParsedExpression],
+                unaryTestsParser: String => Either[String, ParsedExpression]) {
 
   import DmnParser._
 
@@ -432,11 +432,11 @@ class DmnParser(configuration: Configuration,
             var escapedExpression =
               escapeNamesInExpression(expression, ctx.namesToEscape)
 
-            FeelParser.parseUnaryTests(escapedExpression) match {
-              case Success(exp, _) => ParsedExpression(exp, expression)
-              case e: NoSuccess => {
+            unaryTestsParser(escapedExpression) match {
+              case Right(parsedExpression) => parsedExpression
+              case Left(failure) => {
                 ctx.failures += Failure(
-                  s"Failed to parse FEEL unary-tests '$expression':\n$e")
+                  s"Failed to parse FEEL unary-tests '$expression': $failure")
                 ParsedExpression(ConstNull, expression)
               }
             }

--- a/pom.xml
+++ b/pom.xml
@@ -17,12 +17,12 @@
   <name>DMN Scala Root</name>
 
   <properties>
-    <feel.version>1.12.2</feel.version>
-    <camunda.version>7.13.0</camunda.version>
-    <zeebe.version>0.24.4</zeebe.version>
+    <feel.version>1.13.1</feel.version>
+    <camunda.version>7.14.0</camunda.version>
+    <zeebe.version>0.26.1</zeebe.version>
     <version.java>11</version.java>
-    <scala.version>2.13.0</scala.version>
-    <scala.binary.version>2.13.0</scala.binary.version>
+    <scala.version>2.13.5</scala.version>
+    <scala.binary.version>2.13.5</scala.binary.version>
     <encoding>UTF-8</encoding>
     <project.build.sourceEncoding>${encoding}</project.build.sourceEncoding>
     <project.build.resourceEncoding>${encoding}</project.build.resourceEncoding>
@@ -151,25 +151,25 @@
       <dependency>
         <groupId>org.slf4j</groupId>
         <artifactId>slf4j-api</artifactId>
-        <version>1.7.25</version>
+        <version>1.7.30</version>
       </dependency>
 
       <dependency>
         <groupId>org.apache.logging.log4j</groupId>
         <artifactId>log4j-api</artifactId>
-        <version>2.13.3</version>
+        <version>2.14.1</version>
       </dependency>
 
       <dependency>
         <groupId>org.apache.logging.log4j</groupId>
         <artifactId>log4j-core</artifactId>
-        <version>2.13.3</version>
+        <version>2.14.1</version>
       </dependency>
 
       <dependency>
         <groupId>org.apache.logging.log4j</groupId>
         <artifactId>log4j-slf4j-impl</artifactId>
-        <version>2.13.3</version>
+        <version>2.14.1</version>
       </dependency>
 
       <dependency>

